src/bridge/SequencerInbox.sol

```diff
@@ -58,918 +32,1154 @@ import {IEspressoTEEVerifier} from "espresso-tee-contracts/interface/IEspressoTE
     }
-
-    function _chainIdChanged() internal view returns (bool) {
-        return deployTimeChainId != block.chainid;
+    reader4844 = reader4844_;
+    isUsingFeeToken = _isUsingFeeToken;
+    isDelayBufferable = _isDelayBufferable;
+  }
+
+  function _chainIdChanged() internal view returns (bool) {
+    return deployTimeChainId != block.chainid;
+  }
+
+  function postUpgradeInit(
+    BufferConfig memory bufferConfig_
+  ) external onlyDelegated onlyProxyOwner {
+    if (!isDelayBufferable) revert NotDelayBufferable();
+
+    // Assuming we would not upgrade from a version that does not have the buffer initialized
+    // If that is the case, postUpgradeInit do not need to be called
+    if (buffer.bufferBlocks != 0) {
+      revert AlreadyInit();
     }
 
-    function postUpgradeInit() external onlyDelegated onlyProxyOwner {
-        // Assuming we would not upgrade from a version that have MaxTimeVariation all set to zero
-        // If that is the case, postUpgradeInit do not need to be called
-        if (
-            __LEGACY_MAX_TIME_VARIATION.delayBlocks == 0 &&
-            __LEGACY_MAX_TIME_VARIATION.futureBlocks == 0 &&
-            __LEGACY_MAX_TIME_VARIATION.delaySeconds == 0 &&
-            __LEGACY_MAX_TIME_VARIATION.futureSeconds == 0
-        ) {
-            revert AlreadyInit();
-        }
-
-        if (
-            __LEGACY_MAX_TIME_VARIATION.delayBlocks > type(uint64).max ||
-            __LEGACY_MAX_TIME_VARIATION.futureBlocks > type(uint64).max ||
-            __LEGACY_MAX_TIME_VARIATION.delaySeconds > type(uint64).max ||
-            __LEGACY_MAX_TIME_VARIATION.futureSeconds > type(uint64).max
-        ) {
-            revert BadPostUpgradeInit();
-        }
-
-        delayBlocks = uint64(__LEGACY_MAX_TIME_VARIATION.delayBlocks);
-        futureBlocks = uint64(__LEGACY_MAX_TIME_VARIATION.futureBlocks);
-        delaySeconds = uint64(__LEGACY_MAX_TIME_VARIATION.delaySeconds);
-        futureSeconds = uint64(__LEGACY_MAX_TIME_VARIATION.futureSeconds);
-
-        __LEGACY_MAX_TIME_VARIATION.delayBlocks = 0;
-        __LEGACY_MAX_TIME_VARIATION.futureBlocks = 0;
-        __LEGACY_MAX_TIME_VARIATION.delaySeconds = 0;
-        __LEGACY_MAX_TIME_VARIATION.futureSeconds = 0;
-    }
+    _setBufferConfig(bufferConfig_);
+  }
 
-    /**
+  /**
         Deprecated because we created another `initialize` function that accepts the `EspressoTEEVerifier` contract
         address as a parameter which is used by the `SequencerInbox` contract to verify the TEE attestation quote.
      */
-    function initialize(
-        IBridge bridge_,
-        ISequencerInbox.MaxTimeVariation calldata maxTimeVariation_
-    ) external onlyDelegated {
-        revert Deprecated();
+  function initialize(
+    IBridge bridge_,
+    ISequencerInbox.MaxTimeVariation calldata maxTimeVariation_,
+    BufferConfig memory bufferConfig_
+  ) external onlyDelegated {
+    revert Deprecated();
+  }
+
+  function initialize(
```

Comment:
Are we going to need this? I think for migrations to 3.1.0 the rollup will be using entirely new transparent proxies meaning if we don't want this method to exist then we won't need to include it at all instead of deprecating it? Or am I mistaken? CC @Sneh1999